distributed/deploy/tests/test_cluster.py

```diff
@@ -35,3 +36,19 @@ async def test_logs_deprecated():
     async with Cluster(asynchronous=True) as cluster:
         with pytest.warns(FutureWarning, match="get_logs"):
             cluster.logs()
+
+
+@gen_test()
+async def test_deprecated_loop_properties():
+    class ExampleCluster(Cluster):
```

Comment:
nit: It took me a few times reading this to realize that the fact that you were making a subclass wasn't relevant here, just the fact that you were doing `self.loop =`. This and other tests like it might be more readable as
```python
async with Cluster(asynchronous=True, loop=IOLoop.current()) as cluster:
    with pytest.warns(DeprecationWarning) as warninfo:
            cluster.loop = IOLoop.current()
```
(if that actually works)

Reply:
The only usecase of mutating the loop is in subclasses, it made sense to do this because the global event loop exists for the entire `super().__init__` chain, and so sometimes you need to access attributes with the loop in place.

Without a global loop it can only be created when the distributed object is started which can only happen after all the constructors have run.

It never made sense to mutate the cluster.loop after starting it, everything would crash very quickly if someone were to do that

Reply:
Yeah, that makes sense